lisp/ess-mode.el

```diff
@@ -429,32 +421,40 @@ current function."
 
 
 ;;; Formatting / indentation
+(defvar-local ess--installed-style-vars nil
+  "A cons of the form (STYLE . VARS).
+VARS is a list of all style vars which were not set explicitly to
+buffer local values by the user in mode hooks.")
 
-(defun ess-set-style (&optional style quiet)
+(defun ess-set-style (&optional style quiet no-overwrite)
   "Set up the `ess-mode' style variables from the `ess-style' variable.
 If STYLE argument is given, use that instead. It makes the ESS
 indentation style variables buffer local. When non-nil, QUIET
-suppresses messaging."
+suppresses messaging. If NO-OVERWRITE is non-nil, don't overwrite
+style variables which have been set in mode hooks."
   (interactive)
-  (let ((ess-styles (mapcar 'symbol-name (mapcar 'car ess-style-alist))))
-    (unless style
-      (setq style
-            (intern (ess-completing-read "Set ESS mode indentation style"
-                                         ess-styles nil t nil nil ess-style))))
-    (setq style (or style ess-style))
-    (make-local-variable 'ess-style)
-    (if (memq (symbol-name style) ess-styles)
-        (setq ess-style style)
-      (error (format "Bad ESS style: %s" style)))
-    (if (not quiet)
-        (message "ESS-style: %s" ess-style))
-    ;; finally, set the indentation style variables making each one local
-    (mapc (lambda (ess-style-pair)
-            (make-local-variable (car ess-style-pair))
-            (set (car ess-style-pair)
-                 (cdr ess-style-pair)))
-          (cdr (assq ess-style ess-style-alist)))
-    ess-style))
+  (setq style (or style
+                  (let ((styles (mapcar (lambda (x) (symbol-name (car x)))
+                                        ess-style-alist)))
+                    (intern (ess-completing-read
+                             "Set ESS mode indentation style"
+                             styles nil t nil nil ess-style)))))
+  (let ((style-alist (or (cdr (assq style ess-style-alist))
+                         (format "Bad ESS style: %s" style)))
+        (vars (or (cdr ess--installed-style-vars)
+                  (cl-loop for (var . _) in (cdr (assq 'DEFAULT ess-style-alist))
+                           unless (local-variable-p var)
+                           collect var))))
+    (unless quiet
+      (message "ESS-style: %s" style))
+    (mapc (lambda (var)
+            (make-local-variable var)
+            (set var (cdr (assq var style-alist))))
+          (if no-overwrite
+              vars
+            (mapcar 'car style-alist)))
```

Comment:
car should be sharp quoted, right?

Reply:
Right.

Reply:
@jabranham I think I have just by mistake overwritten some of your commits from today in master. Could you please check and force push if so? Have no idea idea how that could have happened but that's is of less importance now.

Reply:
Ok. I think it's a bug in magit. I have just updated it today and now on a non-master branch force-push is pushing to the master. Be careful everyone.

Reply:
We could turn on github's protected branch stuff to disable force-pushing to master. I'd be in favor of that.

Reply:
Yes. Let's do that.